plugins/ipam/dhcp/lease.go

```diff
@@ -202,10 +202,19 @@ func (l *DHCPLease) Stop() {
 
 func (l *DHCPLease) getOptionsWithClientId() dhcp4.Options {
 	opts := make(dhcp4.Options)
-	opts[dhcp4.OptionClientIdentifier] = []byte(l.clientID)
+
+	var clientIdentifier []byte
+	if val, ok := l.optsProviding[dhcp4.OptionClientIdentifier]; ok {
+		clientIdentifier = val
+	} else {
+		// if not set by config, use the default one set by AcquireLease
+		clientIdentifier = []byte(l.clientID)
+	}
+
 	// client identifier's first byte is "type"
-	newClientID := []byte{0}
-	newClientID = append(newClientID, opts[dhcp4.OptionClientIdentifier]...)
+	// 0 means the identifier is not an hardware address
+	// reference: https://datatracker.ietf.org/doc/html/rfc2132#section-9.14
+	newClientID := append([]byte{0}, clientIdentifier...)
```

Comment:
@SilverBut do we want to allow passing the client ID type along with the client ID via the ProvideOptions in the IPAM config? eg, if you use the HostDevice plugin to move a NIC into the container, then want to run DHCP on it, perhaps you do want to use the MAC address as the client ID.

Reply:
Sounds good but I'm worrying this type field will be misused by users, and setting a value not in [the registry](https://www.iana.org/assignments/arp-parameters/arp-parameters.xhtml#arp-parameters-2). 

Instead, maybe we can have an option like "useDeviceMacAsClientIdentifier"?